test/integration/controllermanager/managedseedset/managedseedset_test.go

```diff
@@ -0,0 +1,376 @@
+					RollingUpdate: &seedmanagementv1alpha1.RollingUpdateStrategy{
+						Partition: pointer.Int32(0),
+					},
+				},
+			},
+		}
+
+		shootList = &gardencorev1beta1.ShootList{}
+		seedList = &gardencorev1beta1.SeedList{}
+		managedSeedList = &seedmanagementv1alpha1.ManagedSeedList{}
+		selector, err = metav1.LabelSelectorAsSelector(&managedSeedSet.Spec.Selector)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	JustBeforeEach(func() {
+		By("Create ManagedSeedSet")
+		Expect(testClient.Create(ctx, managedSeedSet)).To(Succeed())
+		log.Info("Created ManagedSeedSet for test", "managedSeedSet", client.ObjectKeyFromObject(managedSeedSet))
+
+		DeferCleanup(func() {
+			By("Delete ManagedSeedSet")
+			Expect(testClient.Delete(ctx, managedSeedSet)).To(Or(Succeed(), BeNotFoundError()))
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(managedSeedSet), managedSeedSet)
+			}).Should(BeNotFoundError())
+		})
+	})
+
+	Context("reconcile", func() {
+		It("should add the finalizer", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedSeedSet), managedSeedSet)).To(Succeed())
+				g.Expect(managedSeedSet.Finalizers).To(ConsistOf("gardener"))
+			}).Should(Succeed())
+		})
+
+		It("should create Shoot from shoot template and set the status.replica value to 1 (default value)", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedSeedSet), managedSeedSet)).To(Succeed())
+				g.Expect(managedSeedSet.Status.Replicas).To(Equal(int32(1)))
+				g.Expect(testClient.List(ctx, shootList, client.InNamespace(managedSeedSet.Namespace), client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(shootList.Items).To(HaveLen(1))
+			}).Should(Succeed())
+		})
+
+		It("should create ManagedSeed when shoot is reconciled successfully", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.List(ctx, shootList, client.InNamespace(managedSeedSet.Namespace), client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(shootList.Items).To(HaveLen(1))
+			}).Should(Succeed())
+
+			By("Mark the Shoot as 'successfully created'")
+			patch := client.MergeFrom(shootList.Items[0].DeepCopy())
+			shootList.Items[0].Status = gardencorev1beta1.ShootStatus{
+				ObservedGeneration: shootList.Items[0].GetGeneration(),
+				LastOperation: &gardencorev1beta1.LastOperation{
+					Type:  gardencorev1beta1.LastOperationTypeCreate,
+					State: gardencorev1beta1.LastOperationStateSucceeded,
+				},
+			}
+			Expect(testClient.Status().Patch(ctx, &shootList.Items[0], patch)).To(Succeed())
+
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.List(ctx, managedSeedList, client.InNamespace(managedSeedSet.Namespace), client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(managedSeedList.Items).To(HaveLen(1))
```

Comment:
Should we do some more assertions here on the `ManagedSeed` object to make sure the "correct object" was created?